Sources/OktaIdx/IDXToken.swift

```diff
@@ -58,29 +57,81 @@ extension IDXClient {
             }
             
             guard let tokenString = selectedToken else {
-                completion(false, IDXClientError.invalidParameter(name: "token"))
+                completion(.failure(IDXClientError.invalidParameter(name: "token")))
                 return
             }
             
             Token.revoke(token: tokenString, type: type, configuration: configuration, completion: completion)
         }
 
+        /// Revokes the token.
+        /// - Parameters:
+        ///   - type: The type to revoke (e.g. access token, or refresh token).
+        ///   - completion: Completion handler for when the token is revoked.
+        @objc(revokeToken:completion:)
+        public func revoke(type: Token.RevokeType = .accessAndRefreshToken, completion: @escaping(_ successful: Bool, _ error: Error?) -> Void) {
+            revoke(type: type) { result in
+                switch result {
+                case .success(_):
+                    completion(true, nil)
+                case .failure(let error):
+                    completion(false, error)
+                }
+            }
+        }
+
         /// Revokes the given token using the string value of the token.
         /// - Parameters:
         ///   - token: Token string to revoke.
         ///   - type: The type to revoke (e.g. access token, or refresh token).
         ///   - configuration: The client configuration used when the token was created.
         ///   - completion: Completion handler for when the token is revoked.
-        @objc(revokeToken:type:configuration:completion:)
         public static func revoke(token: String,
                                   type: Token.RevokeType,
                                   configuration: Configuration,
-                                  completion: @escaping(_ successful: Bool, _ error: Error?) -> Void)
+                                  completion: @escaping(Result<Void,IDXClientError>) -> Void)
         {
             let api = IDXClient.Version.latest.clientImplementation(with: configuration)
             revoke(token: token, type: type, api: api, completion: completion)
         }
+
+        /// Revokes the given token using the string value of the token.
+        /// - Parameters:
+        ///   - token: Token string to revoke.
+        ///   - type: The type to revoke (e.g. access token, or refresh token).
+        ///   - configuration: The client configuration used when the token was created.
+        ///   - completion: Completion handler for when the token is revoked.
+        @objc(revokeToken:type:configuration:completion:)
+        public static func revoke(token: String,
+                                  type: Token.RevokeType,
+                                  configuration: Configuration,
+                                  completion: @escaping(_ successful: Bool, _ error: Error?) -> Void)
```

Comment:
Shouldn't we return `Result` here as well?

Reply:
Not on the @objc methods. Objective-C doesn't support enums with associated values, so there are 2 varieties of every method: 1) Swift-only methods that support `Result<T, IDXClientError>`, and 2) Objective-C-compatible methods that accept two arguments. The conventional completion blocks can still be used from Swift code as well, for those developers that don't want to use the result type.